src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -187,53 +236,93 @@ public async Task<TrialResult> RunAsync(CancellationToken ct = default)
             ValidateSettings();
             Initialize();
             _settings.CancellationToken = ct;
-            _cts.CancelAfter((int)_settings.MaxExperimentTimeInSeconds * 1000);
-            _settings.CancellationToken.Register(() => _cts.Cancel());
+            _globalCancellationTokenSource.CancelAfter((int)_settings.MaxExperimentTimeInSeconds * 1000);
```

Comment:
>)_settings.MaxExperimentTimeInSeconds * 1000 [](http://example.com/codeflow?start=59&length=44)

Any concern with overflowing this calculation?

Reply:
Thanks for catching that, switch to `TimeSpan.FromSeconds` to avoid overflow in int

Reply:
### Resolved